clevels/austerity.go

```diff
@@ -127,6 +128,26 @@ func Criticality(line string) AusterityLevel {
 	return DefaultCriticality
 }
 
+func JSONCriticality(line json.LogLine) AusterityLevel {
+	// Never drop JSON log lines declaring themselves canonical:
+	if canonicalI, ok := line["canonical"]; ok {
```

Comment:
do we have to include this here? I think we managed to keep `CANONICAL` out of the OSS project for the non-JSON logs, and it'd be great if we could do that for JSON too. These filters can be configured/overridden, so we just need to add it to the compiled and deployed version instead.

Reply:
It mirrors what we do for text-based logs in OSS unilog: The `canonical-.*line` regexp has been in here for about 1.5 years https://github.com/stripe/unilog/blob/3e8ab1010b96c63b9e309764b43e3bfb4534adee/clevels/austerity.go#L97 (:

Reply:
Without even checking, I can pretty much guess then that that git-blames to me, in which case I literally only have myself to blame. 🙃

Reply:
😂 yep I'm pretty sure it does ((:

Reply:
Sooo we need to fix this regex because it won't currently match `CANONICAL-FOO-BAR-LINE`

Reply:
This PR doesn't touch that regex - I can change it here, but seems better to do in a separate PR. I'll open another one.

Reply:
Made #40 for this.